Makefile

```diff
@@ -0,0 +1,9 @@
+# Do things in edx-platform
+
+clean:
+	# Remove all the git-ignored stuff, but save and restore things marked
+	# by start-noclean/end-noclean.
+	sed -n -e '/start-noclean/,/end-noclean/p' < .gitignore > /tmp/private-files
+	tar cf /tmp/private.tar `git ls-files --exclude-from=/tmp/private-files --ignored --others`
+	git clean -fdX
+	tar xf /tmp/private.tar
```

Comment:
Maybe use the method suggested here: http://stackoverflow.com/questions/11085795/git-exclude-a-file-with-git-clean

In particular, use `-x`, in combination with explicit `-e`s for the developer files we want to keep. Although... I guess that would delete manually created files, which is no good.